agent/containermetadata/metadata_manager_unix.go

```diff
@@ -0,0 +1,80 @@
+// +build !windows
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerclient"
+)
+
+const (
+	mountPoint                     = "/ecs/metadata"
+	ContainerMetadataClientVersion = dockerclient.Version_1_21
+)
+
+// CreateMetadata creates the metadata file and adds the metadata directory to
+// the container's mounted host volumes
+// binds []string is passed by value to avoid race conditions by multiple
+// calls to CreateMetadata, although this should never actually happen
+func (manager *metadataManager) CreateMetadata(binds []string, task *api.Task, container *api.Container) ([]string, error) {
```

Comment:
My comments on previous revision seem to have disappeared (probably because you renamed the file), but I'd still like them addressed.